### DISCUSSION TITLE
Fullstack .NET - uso do identity no blazor

### DISCUSSION THREAD

**Discussion by gabrieladouradof** [balta-io/forum#704] [Tira Dúvidas] [answered]
Bom dia pessoal! Estou empacada há alguns dias em um problema que acredito que tenha sido durante a integração do Identity no blazor, pois quando implementei o identity apenas na API ela funcionava normalmente. Hoje quando insiro meus dados na página já web para criar cadastros recebo esse erro via ferramentas do desenvolvedor: registrar:1 Access to fetch at 'http://localhost:5252/v1/identity/register' from origin 'http://localhost:5181' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

![image](https://github.com/user-attachments/assets/b59d25fd-c08a-4e12-b7db-20c060c51b70)

Gostaria de entender mais.

**Comment by atilionogueira** [comment, marked as answer]
1 - No método AddCrossOrigin veja se as portas do appsettings.json  estão corretas
    Configuration.BackendUrl
    Configuration.FrontendUrl

appsettings.json
{
  "Configuration": {
    "BackendUrl": "http://localhost:XXXX",
    "FrontendUrl": "http://localhost:XXXX"
  }
}

2 - No arquivo program.cs Dima.Core veja se o app.UserCors está sequencia abaixo:

var app = builder.Build();

if (app.Environment.IsDevelopment())
    app.ConfigureDevEnvironment();

app.UseCors(ApiConfiguration.CorsPolicyName);
app.UseSecurity();
app.MapEndpoints();


Se após essas verificações o problema persistir, compartilhe o repositório do GitHub.

**Reply by gabrieladouradof** [reply]
Obrigada, Atilio, pela ajuda! Infelizmente não resolveu, e minhas URLs estao corretas. Este é meu repositório: https://github.com/gabrieladouradof/ApplicationFullstack.NET

**Reply by andrebaltieri** [reply]
@gabrieladouradof durante a execução do programa, seu ApiConfiguration.CorsPolicyName está trazendo o valor correto? E as configurações de URL?

Pelo que vi aqui, a configuração do CORS em si está OK!
Debugando você se deparou com algum erro?

**Reply by gabrieladouradof** [reply]
@andrebaltieri então, ele traz o valor wasm, que definimos, entretanto notei que as variáveis de URL estão recebendo valores nulos: 
![Imagem do WhatsApp de 2024-11-25 à(s) 15 09 04_c8f9fc4b](https://github.com/user-attachments/assets/a31e8bcb-eca3-4c3d-9dcd-fa30f18eec04) 
![Imagem do WhatsApp de 2024-11-25 à(s) 15 09 36_5b537e8c](https://github.com/user-attachments/assets/2a669534-280f-4aab-911a-95d6020349ed)
![Imagem do WhatsApp de 2024-11-25 à(s) 15 10 07_e2157e20](https://github.com/user-attachments/assets/dd0de062-958b-4991-96e5-51cc42366f68)

**Reply by malighieri** [reply]
Vi no seu appsettings que a configuração das URLs parece estar dentro de chaves que não deveria. Pode ser isso que está impedindo das configurações serem carregadas corretamente

![image](https://github.com/user-attachments/assets/dd070707-af40-4756-a369-891a037dc5b6)

**Reply by gabrieladouradof** [reply]
Olá Miron, muito obrigada pelo palpite! eu arrumei, mas infelizmente continuo com o omesmo problema. :(

**Comment by tiaotj** [comment]
Olá, Gabriela
O meu está com o mesmo problema, vc já conseguiu alguma solução?

**Comment by tiaotj** [comment]
@gabrieladouradof acredito que seu problema é o mesmo que o meu, consegui resolver. observa o seguinte
![image](https://github.com/user-attachments/assets/10061bd8-6ea8-45f6-b963-0a7c9b5f0a50)
A origem no erro, está http
e vc configurou https
![image](https://github.com/user-attachments/assets/6b1f24c2-83a9-4e09-8cda-185d9739c9a0)
Acredito que mudando isso já vai funcionar

**Reply by gabrieladouradof** [reply]
Olá @tiaotj, meu problema era na configuração do CORS, não entendi muito bem a razão, mas eu resolvi re-configurando, vi que algumas configurações não se faziam necessárias, para este projeto. Notei que você também resolveu seu problema, mandou bem! 
![image](https://github.com/user-attachments/assets/c40f29a0-79eb-4c5e-89f3-1b1658280e18)
